usr/libexec/helper-scripts/accountctl.sh

```diff
@@ -0,0 +1,395 @@
+    "!"*) return 0;;
+    *) return 1;;
+  esac
+}
+
+
+## Description: Check if password is disabled.
+## Output: None
+## Return: 0 when disabled, 1 otherwise.
+## Usage: is_pass_disabled USER
+## Example: is_pass_disabled user
+is_pass_disabled(){
+  log info "${FUNCNAME[0]} $*"
+  local user
+  user="${1:-}"
+  is_user "${user}"
+  case "$(get_pass "${user}")" in
+    "*"*|"!*"*) return 0;;
+    *) return 1;;
+  esac
+}
+
+
+## Description: Lock a password.
+## Output: None
+## Return: passwd
+## Usage: lock_pass USER
+## Example: lock_pass user
+lock_pass(){
+  log info "${FUNCNAME[0]} $*"
+  has passwd
+  local user
+  user="${1:-}"
+  is_user "${user}"
+  if is_pass_locked "${user}"; then
+    return 0
+  fi
+  passwd -ql -- "${user}"
+}
+
+
+## Description: Unlock a password.
+## Output: None
+## Return: chpasswd
+## Usage: unlock_pass USER
+## Example: unlock_pass user
+unlock_pass(){
+  log info "${FUNCNAME[0]} $*"
+  has chpasswd
+  local user
+  user="${1:-}"
+  is_user "${user}"
+  if ! is_pass_locked "${user}"; then
+    return 0
+  fi
+  ## The tools "passwd" and "usermod" can't unlock when password is empty.
+  pass="$(get_clean_pass "${user}" '!')"
+  chpasswd -c NONE <<< "${user}:${pass}"
+}
+
+
+## Description: Disable a password.
+## Output: None
+## Return: chpasswd
+## Usage: disable_pass USER
+## Example: disable_pass user
+disable_pass(){
+  log info "${FUNCNAME[0]} $*"
+  has chpasswd
+  local user
+  user="${1:-}"
+  is_user "${user}"
+  if is_pass_disabled "${user}"; then
+    return 0
+  fi
+  pass="$(get_pass "${user}")"
+  chpasswd -c NONE <<< "${user}:*${pass}"
+}
+
+
+## Description: Enable a password.
+## Output: None
+## Return: chpasswd
+## Usage: enable_pass USER
+## Example: enable_pass user
+enable_pass(){
+  log info "${FUNCNAME[0]} $*"
+  has chpasswd
+  local user
+  user="${1:-}"
+  is_user "${user}"
+  if ! is_pass_disabled "${user}"; then
+    return 0
+  fi
+  if is_pass_locked "${user}"; then
+    pass="$(get_clean_pass "${user}" '!*')"
+    chpasswd -c NONE <<< "${user}:!${pass}"
+    return 0
+  fi
+  pass="$(get_clean_pass "${user}" '*')"
+  chpasswd -c NONE <<< "${user}:${pass}"
+}
+
+
+## Description: Transform field name to index per database
+## Output: Field 0-indexed
+## Return: Exit 1 if entry does not exist
+## Usage: get_field DB FIELD
+## Example: get_field passwd pass
+get_field(){
+  log info "${FUNCNAME[0]} $*"
+  local db field index
+  db="${1:-}"
+  field="${2:-}"
+  index=""
+  case "${db}" in
+    passwd)
+      case "${field}" in
+        pass) index=2;;
+        uid) index=3;;
+        gid) index=4;;
+        comment) index=5;;
+        home) index=6;;
+        shell) index=7;;
+      esac
+      ;;
+    shadow)
+      case "${field}" in
+        pass) index=2;;
+        last-pass-change) index=3;;
+        min-pass-age) index=4;;
+        max-pass-age) index=5;;
+        warn-pass-period) index=6;;
+        lock-pass-period) index=7;;
+        expiration-date) index=7;;
```

Comment:
I missed this last time, but I think this index is wrong.

```suggestion
        expiration-date) index=8;;
```